src/Microsoft.AspNet.Mvc.Xml/RequiredValidationHelper.cs

```diff
@@ -0,0 +1,182 @@
+                        {
+                            errorMessages = new List<string>();
+                            errors.Add(validationError.Value.ModelType, errorMessages);
+                        }
+
+                        errorMessages.Add(Resources.FormatRequiredProperty_MustHaveDataMemberRequired(
+                                        validationError.Value.PropertyName,
+                                        validationError.Value.ModelType.FullName));
+                    }
+                    
+                    // if the type is not primitve, then it could be a struct in which case
+                    // we need to probe its properties for validation
+                    if (propertyType.GetTypeInfo().IsPrimitive)
+                    {
+                        continue;
+                    }
+                }
+
+                var childNodeErrors = new Dictionary<Type, List<string>>();
+                Validate(propertyType, visitedTypes, childNodeErrors);
+
+                // Avoid adding duplicate errors at current node.
+                foreach (var modelTypeKey in childNodeErrors.Keys)
+                {
+                    if (!errors.ContainsKey(modelTypeKey))
+                    {
+                        errors.Add(modelTypeKey, childNodeErrors[modelTypeKey]);
+                    }
+                }
+            }
+
+            cachedValidationErrors.TryAdd(modelType, errors);
+
+            visitedTypes.Remove(modelType);
+        }
+
+        private ValidationError? GetValidationError(PropertyInfo propertyInfo)
+        {
+            var required = propertyInfo.GetCustomAttribute(typeof(RequiredAttribute), inherit: true);
+            if (required == null)
+            {
+                return null;
+            }
+
+            var dataMemberRequired = (DataMemberAttribute)propertyInfo.GetCustomAttribute(
+                typeof(DataMemberAttribute),
+                inherit: true);
+
+            if (dataMemberRequired != null && dataMemberRequired.IsRequired)
+            {
+                return null;
+            }
+
+            return new ValidationError()
+            {
+                ModelType = propertyInfo.DeclaringType,
+                PropertyName = propertyInfo.Name
+            };
+        }
+
+        private bool ExcludeTypeFromValidation(Type modelType)
+        {
+            return modelType.GetTypeInfo().IsPrimitive ||
+                            modelType.Equals(typeof(decimal)) ||
+                            modelType.Equals(typeof(string)) ||
+                            modelType.Equals(typeof(DateTime)) ||
+                            modelType.Equals(typeof(Guid)) ||
+                            modelType.Equals(typeof(DateTimeOffset)) ||
+                            modelType.Equals(typeof(TimeSpan)) ||
+                            modelType.Equals(typeof(Uri));
+        }
```

Comment:
yeah, this appears here : https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNet.Mvc.ModelBinding/SimpleTypesExcludeFilter.cs#L55

Considered this but was not sure if I should bring in the exclude filters into xmlformatter.